src/ripple/proto/org/xrpl/rpc/v1/xrp_ledger.proto

```diff
@@ -4,39 +4,18 @@ package org.xrpl.rpc.v1;
 option java_package = "org.xrpl.rpc.v1";
 option java_multiple_files = true;
 
-import "org/xrpl/rpc/v1/get_account_info.proto";
-import "org/xrpl/rpc/v1/get_fee.proto";
-import "org/xrpl/rpc/v1/submit.proto";
-import "org/xrpl/rpc/v1/get_transaction.proto";
-import "org/xrpl/rpc/v1/get_account_transaction_history.proto";
 import "org/xrpl/rpc/v1/get_ledger.proto";
 import "org/xrpl/rpc/v1/get_ledger_entry.proto";
 import "org/xrpl/rpc/v1/get_ledger_data.proto";
 import "org/xrpl/rpc/v1/get_ledger_diff.proto";
 
 
-// RPCs available to interact with the XRP Ledger.
-// The gRPC API mimics the JSON API. Refer to xrpl.org for documentation
+// These methods are binary only methods for retrieiving arbitrary ledger state
+// via gRPC. These methods are used by clio and reporting mode, but can also be
+// used by any client that wants to extract ledger state in an efficient manner.
+// They do not directly mimic the JSON equivalent methods.
 service XRPLedgerAPIService {
 
-  // Get account info for an account on the XRP Ledger.
-  rpc GetAccountInfo (GetAccountInfoRequest) returns (GetAccountInfoResponse);
-
-  // Get the fee for a transaction on the XRP Ledger.
-  rpc GetFee (GetFeeRequest) returns (GetFeeResponse);
-
-  // Submit a signed transaction to the XRP Ledger.
-  rpc SubmitTransaction (SubmitTransactionRequest) returns (SubmitTransactionResponse);
-
-  // Get the status of a transaction
-  rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse);
-
-  // Get all validated transactions associated with a given account
-  rpc GetAccountTransactionHistory(GetAccountTransactionHistoryRequest) returns (GetAccountTransactionHistoryResponse);
-
-  /////////////////////////////////////////////////////////////////////////////
-  // The below methods do not mimic the JSON API exactly, and are mostly binary
-
   // Get a specific ledger, optionally including transactions and any modified,
   // added or deleted ledger objects
   rpc GetLedger(GetLedgerRequest) returns (GetLedgerResponse);
```

Comment:
Can we also remove `GetLedgerDiff()` and `GetLedgerEntry`? AFAIK, Clio only uses `GetLedgerRequest` and `GetLedgerDataRequest`

Reply:
Those RPCs are not part of the xpring sdk, and I actually think they could be useful. If we want to remove them, we should discuss and do it as part of a separate PR.